appinfo/routes.php

```diff
@@ -25,7 +25,7 @@
 		//Retrieve Metadata of a list of files
 		[
 			'name' => 'metadata#get_metadata',
-			'url'  => '/metadata/{fileList}',
+			'url'  => '/metadata?fileList={fileList}',
```

Comment:
I was mostly looking at the code examples, but it's true that if you don't have prior experience and reading the text, then there is not enough information in there to fully understand how it works. The Symfony doc is not doing a better job either.
Let's not waste any more time.
You don't need to add anything to the URL in your route. GET parameters are automatically transmitted to the controller. You only need to add them to the URL if you send them in your URL (`/ap/preview/id/width/height` per example)

So use something like this:

```
                [
                        'name' => 'metadata_api#get_metadata',
                        'url'  => '/api/metadata',
                        'verb' => 'GET'
                ],
```

Since you're expecting a string that you will convert to an array, you don't need to cast the parameter in your controller, but it's sometimes necessary, when you send booleans as GET params per example.